examples/examples/toggle-layers.rs

```diff
@@ -0,0 +1,47 @@
+#![deny(rust_2018_idioms)]
+/// This is a example showing how `Layer` can be enabled or disabled by
+/// by wrapping them inside a `Option`. This example shows `fmt` and `json`
+/// being toggled based on the `json` command line flag.
+///
+/// You can run this example by running the following command in a terminal
+///
+/// ```
+/// cargo run --example toggle-layers -- json
+/// ```
+///
+use clap::{App, Arg};
+use tracing::info;
+use tracing_subscriber::{prelude::__tracing_subscriber_SubscriberExt, util::SubscriberInitExt};
+
+#[path = "fmt/yak_shave.rs"]
+mod yak_shave;
+
+fn main() {
+    let matches = App::new("fmt optional Example")
+        .version("1.0")
+        .arg(
+            Arg::with_name("json")
+                .help("Enabling json formatting of logs")
+                .required(false)
+                .takes_value(false),
+        )
+        .get_matches();
+
+    let (json, plain) = if matches.is_present("json") {
+        (Some(tracing_subscriber::fmt::layer().json()), None)
+    } else {
+        (None, Some(tracing_subscriber::fmt::layer()))
+    };
+
+    tracing_subscriber::registry().with(json).with(plain).init();
```

Comment:
It would be nice to have an abbreviated version of this example in the docs for the `Layer` trait. Maybe also a version showing runtime reloading in the `layer::reload` docs.